internal/handlers/handler_test.go

```diff
@@ -124,11 +125,47 @@ func TestFind(t *testing.T) {
 		req     *types.Document
 		resp    *types.Array
 		err     error
+		deep    bool
 	}
 
 	// Do not use sentences, spaces, or underscores in subtest names
 	// to make it easier to run individual tests with `go test -run test/name` and for consistency.
 	testCases := map[string]testCase{
+		// db.values.find({ value: { code: {$gt: 121080 }}}, { value: { $elemMatch: { code: 121082 }}})
+		// output:
+		// elemmatch5 w/o name field, w/o value field
+		// elemmatch6 w/o name field, with value field
+		"elemMatchFilterProjection": {
+			deep:    true,
+			schemas: []string{"values"},
+			req: types.MustNewDocument(
+				"find", "values",
+				"filter", types.MustNewDocument(
+					"value.code", types.MustNewDocument("$gt", int32(121081)),
+				),
+				"sort", types.MustNewDocument(
+					"name", int32(1),
+				),
+				"projection", types.MustNewDocument(
+					"value", types.MustNewDocument(
+						"$elemMatch", types.MustNewDocument(
+							"code", int32(121082),
+						),
+					),
+				),
+			),
+			resp: types.MustNewArray(
+				types.MustNewDocument(
+					"_id", types.ObjectID{0x61, 0x2e, 0xc2, 0x80, 0x00, 0x00, 0x04, 0x05, 0x00, 0x00, 0x03, 0x09},
+				),
+				types.MustNewDocument(
+					"_id", types.ObjectID{0x61, 0x2e, 0xc2, 0x80, 0x00, 0x00, 0x04, 0x05, 0x00, 0x00, 0x03, 0x09},
+					"value", types.MustNewArray(
+						types.MustNewDocument("code", int32(121081), "document", "zyx"),
```

Comment:
found a bug in filter, it should not work like that. 
filter for value.code is gt 121081, but id did returned 121081.

Reply:
for this PR I've changed tests to make it work for `$elemMatch` (code updated so the comment above is not actual).
but in future, it's better to find the find bug and fix.